src/zinoargus/__init__.py

```diff
@@ -277,6 +278,47 @@ def synchronize_continuously(argus_incidents: IncidentMap, zino_cases: CaseMap):
             synchronize_case_history(case, incident)
 
 
+def refresh_argus_incidents(argus_incidents: IncidentMap, zino_cases: CaseMap):
+    """Refreshes all the Argus incidents that we know of and care about.
+
+    This also triggers actions that modify Zino cases if indicated by changes to the
+    Argus incidents.
+    """
+    for case_id, old_incident in argus_incidents.items():
+        new_incident = _argus.get_incident(old_incident.pk)
+        argus_incidents[case_id] = new_incident
+
+        if new_incident.acked and not old_incident.acked:
+            update_case_acknowledged(case_id, argus_incidents, zino_cases)
+
+
+def update_case_acknowledged(
+    case_id: int, argus_incidents: IncidentMap, zino_cases: CaseMap
+):
+    """Updates a Zino case with the acknowledged status from Argus, if necessary."""
+    incident = argus_incidents[case_id]
+    msg = "Argus incident %s (zino case %s) was acknowledged"
+    desired_state = _config.sync.acknowledge.setstate
+    if desired_state == "none":
+        _logger.debug(msg + ", ignoring as configured", incident.pk, case_id)
```

Comment:
I think we can opt out of this earlier, because it will go into this function so often, I think we can already in `synchronize_continuously` check if we're supposed to sync acknowledgements back

Because these log messages would be quite annoying if you configure it to be ignored

Reply:
That's a fair point :)